src/helpers.rkt

```diff
@@ -9,9 +9,9 @@
 
 (define (is-arrow-type? expr)
   (and (list? expr)
-       (equal? (first expr) '->)))
+       (eq? (first expr) '->)))
```

Comment:
Ah, sorry, you're right. `eq?` might still be marginally faster but it's probably not worth the non-uniformity of equality fns this creates.